data_prep/generate_annual_averages.py

```diff
@@ -0,0 +1,191 @@
+import os, os.path
+import re
+import logging
+import sys
+
+from argparse import ArgumentParser
+from datetime import datetime
+
+from cdo import Cdo
+from netCDF4 import Dataset, num2date, date2num
+import numpy as np
+from dateutil.relativedelta import relativedelta
+
+from Cmip5File import Cmip5File
+
+log = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=4)
+
+def iter_matching(dirpath, regexp):
+    # http://stackoverflow.com/questions/4639506/os-walk-with-regex
+    """Generator yielding all files under `dirpath` whose absolute path
+       matches the regular expression `regexp`.
+       Usage:
+
+           >>> for filename in iter_matching('/', r'/home.*\.bak'):
+           ....    # do something
+    """
+    for dir_, dirnames, filenames in os.walk(dirpath):
+        for filename in filenames:
+            abspath = os.path.join(dir_, filename)
+            if regexp.match(abspath):
+                yield abspath
+
+def create_annual_avg_file(in_fp, out_fp, variable):
+    '''
+    Generates file with annual averages for the given variable across
+    all time within an input file
+
+    Parameters:
+        in_fp: input file path
+        out_fp: output file path
+        variable (str): name of the variable which is being processed
+
+    '''
+    supported_vars = {
```

Comment:
The script is irrelevant to all of the ClimDEX variables since they are all defined as being annual.